internal/project/auto/golang.go

```diff
@@ -8,40 +8,92 @@ import (
 	"fmt"
 	"io"
 	"os"
-	"path"
 	"path/filepath"
+	"strings"
 
 	"golang.org/x/mod/modfile"
 
+	"github.com/siderolabs/kres/internal/dag"
 	"github.com/siderolabs/kres/internal/project/common"
 	"github.com/siderolabs/kres/internal/project/golang"
+	"github.com/siderolabs/kres/internal/project/meta"
 	"github.com/siderolabs/kres/internal/project/service"
 	"github.com/siderolabs/kres/internal/project/wrap"
 )
 
 // DetectGolang checks if project at rootPath is Go-based project.
-//
-//nolint:gocognit,gocyclo,cyclop
 func (builder *builder) DetectGolang() (bool, error) {
-	gomodPath := filepath.Join(builder.rootPath, "go.mod")
+	lookupDirs := []string{}
 
-	gomod, err := os.Open(gomodPath)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return false, nil
+	err := filepath.Walk(builder.rootPath, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
 		}
 
+		if strings.HasSuffix(path, "go.mod") {
```

Comment:
this is wrong, better:

```suggestion
		if filepath.Base(path) == "go.mod" {
```